### ISSUE TITLE
question about training code.

### ISSUE THREAD

**Issue by 85zhanghao** [grip-unina/TruFor#11]
Do you plan to upload the training code? I still have some doubts about some training details.

**Comment by fabrizioguillaro** [comment]
Yes, we intend to make the training code available, but we don't have a release date for it yet.
Meanwhile, if you have doubts on training details, feel free to ask.

**Comment by Woodyet** [comment]
Any updates on this?

**Comment by csjunjun** [comment]
Any updates on the training code?

**Comment by Samarium-Z** [comment]
Your work is fascinating! Could you kindly share the training code? Thank you so much!

**Comment by nataliameira** [comment]
Congratulations for your work. Do you have any updates on this?

**Comment by wenxian-ok** [comment]
Your job is very interesting. When do you plan to release the complete code?

**Comment by kartik3ya** [comment]
any updates on training code / pretrained model weights?

**Comment by fabrizioguillaro** [comment]
Training code is now available! It includes training of localization network, detection network and confidence estimator. 
Currently, it does not include Noiseprint++ training, but the Noiseprint++ weights are in `TruFor_train_test/pretrained_models/noiseprint++/` folder. They are also included inside the final TruFor trained weights, as they are kept frozen during training.
